edalize/cocotb.py

```diff
@@ -0,0 +1,157 @@
+import logging
+import os
+import pathlib
+
+import cocotb
+
+from edalize import get_edatool
+
+logger = logging.getLogger(__name__)
+
+
+# This class could be replaced by calls to cocotb-config. Knowledge of these
+# simulator arguments belongs in the cocotb codebase.
+class CocotbConfig:
+
+    @staticmethod
+    def get_simulator_args(simulator, toplevel_lang):
+        if simulator == 'ghdl':
+            return CocotbConfig.get_ghdl_args(toplevel_lang)
+        elif simulator == 'icarus':
+            return CocotbConfig.get_icarus_args(toplevel_lang)
+        else:
+            raise RuntimeError('Simulator ' + simulator + ' is not supported by cocotb backend')
+
+    @staticmethod
+    def get_ghdl_args(toplevel_lang):
+        if toplevel_lang != 'vhdl':
+            raise RuntimeError('GHDL only supports VHDL')
+
+        ghdl_cocotb_lib = pathlib.Path(os.path.dirname(cocotb.__file__)) / 'libs' / 'libcocotbvpi_ghdl.so'
```

Comment:
This won't support windows. Obviously that's cocotb's problem to handle with your proposed strategy though.

Does edalize support windows anyway?

Reply:
Yes, it does. The tests run on Windows. So it will need fixing up one way or another - preferably in cocotb.